plugins/inputs/libvirt/libvirt.go

```diff
@@ -0,0 +1,146 @@
+package libvirt
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const sampleConfig = `
+  ## specify a libvirt connection uri, see https://libvirt.org/uri.html
+  uri = "qemu:///system"
+`
+
+type Libvirt struct {
+	Uri   string
+	virsh Virsh
+}
+
+type Virsh func(uri string, args ...string) (string, error)
+
+func (l *Libvirt) SampleConfig() string {
+	return sampleConfig
+}
+
+func (l *Libvirt) Description() string {
+	return "Read domain infos from a libvirt deamon"
+}
+
+func (l *Libvirt) Gather(acc telegraf.Accumulator) error {
+	domains, err := l.listDomains()
+
+	if err != nil {
+		return err
+	}
+
+	for _, domain := range domains {
+		l.gatherDomain(acc, domain)
+	}
+
+	return nil
+}
+
+func (l *Libvirt) listDomains() ([]string, error) {
+	out, err := l.virsh(l.Uri, "list")
```

Comment:
Should we `virsh list --all`?  Could also use `virsh list --all --name` if we are not pulling state from here.